src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/vector/elements.hpp

```diff
@@ -0,0 +1,265 @@
+    /// @brief Read the entire byte, as a std::uint8_t.
+    constexpr operator std::uint8_t() const noexcept {
+        return *byte;
+    }
+
+    /// @brief Overwrite the usable portion of the byte, and set reserved bits to zero.
+    /// @param v Byte to write. Reserved bits are ignored.
+    /// @return *this
+    packed_bit_byte const& operator=(std::uint8_t const& v) const noexcept {
+        *byte = v & mask;
+        return *this;
+    }
+
+    /// Copy the referenced value from another reference of the same type
+    packed_bit_byte const& operator=(packed_bit_byte const& v) const noexcept {
+        return *this = std::uint8_t(v);
+    }
+
+    /// Operator +=, emulating number reference behavior
+    packed_bit_byte const& operator+=(std::uint8_t const& other) const noexcept {
+        return *this = std::uint8_t(*this + other);
+    }
+
+    /// Operator -=, emulating number reference behavior
+    packed_bit_byte const& operator-=(std::uint8_t const& other) const noexcept {
+        return *this = std::uint8_t(*this - other);
+    }
+
+    /// Operator *=, emulating number reference behavior
+    packed_bit_byte const& operator*=(std::uint8_t const& other) const noexcept {
+        return *this = std::uint8_t(*this * other);
+    }
+
+    /// Operator /=, emulating number reference behavior
+    packed_bit_byte const& operator/=(std::uint8_t const& other) const noexcept {
+        return *this = *this / other;
+    }
+
+    /// Operator %=, emulating number reference behavior
+    packed_bit_byte const& operator%=(std::uint8_t const& other) const noexcept {
+        return *this = *this % other;
+    }
+
+    /// Operator ^=, emulating number reference behavior
+    packed_bit_byte const& operator^=(std::uint8_t const& other) const noexcept {
+        return *this = *this ^ other;
+    }
+
+    /// Operator &=, emulating number reference behavior
+    packed_bit_byte const& operator&=(std::uint8_t const& other) const noexcept {
+        return *this = *this & other;
+    }
+
+    /// Operator |=, emulating number reference behavior
+    packed_bit_byte const& operator|=(std::uint8_t const& other) const noexcept {
+        return *this = *this | other;
+    }
+
+    /// Operator <<=, emulating number reference behavior
+    packed_bit_byte const& operator<<=(unsigned other) const noexcept {
+        return *this = *this << other;
+    }
+
+    /// Operator >>=, emulating number reference behavior
+    packed_bit_byte const& operator>>=(unsigned other) const noexcept {
+        return *this = *this >> other;
+    }
+
+    constexpr packed_bit_byte(packed_bit_byte const& other) : byte(other.byte), mask(other.mask) {}
+
+   private:
+    friend class iterators::packed_bit_byte<Iterator>;
+
+    constexpr packed_bit_byte(Iterator byte_iter, uint8_t byte_mask) noexcept : byte(byte_iter), mask(byte_mask) {}
+
+    Iterator byte;
+    std::uint8_t mask;
+};
+
+/// packed_bit_byte is Swappable even when it's not an lvalue reference
```

Comment:
```suggestion
/// Swap the referenced values for `a` and `b`.
///
/// @note `packed_bit_byte<Iterator>` is a proxy reference and behaves like an lvalue reference.
```
Minor wording suggestions to clarify the expected behavior of `packed_bit_byte<Iterator>` as a "proxy reference" type.